staging/src/k8s.io/apiserver/pkg/endpoints/installer.go

```diff
@@ -159,6 +159,10 @@ func getStorageVersionKind(storageVersioner runtime.GroupVersioner, storage rest
 // object. If the storage object is a subresource and has an override supplied for it, it returns
 // the group version kind supplied in the override.
 func GetResourceKind(groupVersion schema.GroupVersion, storage rest.Storage, typer runtime.ObjectTyper) (schema.GroupVersionKind, error) {
```

Comment:
storage must not be nil here... do you have an example of how you constructed/called this function? I'm not seeing a caller that could reasonably have a nil storage when calling this

Reply:
The call stack was shown in issue https://github.com/kubernetes/kubernetes/issues/109336.
From the stack, we can see the calling chain is:
```
genericapiserver.InstallAPIGroups -> installAPIResources -> endpoints/groupversion.InstallREST -> 
endpoints/installer.Install -> registerResourceHandlers -> GetResourceKind -> storage.New()
```

When the `GetResourceKind` was called , it paniced in `storage.New()`.

Reply:
@liggitt PTAL thanks.

Reply:
I still think that indicates a programming error in the caller… how did a nil storage value get into that map?

Reply:
Checked again, it is the nil map value of `APIGroupInfo.VersionedResourcesStorageMap` (initiated by our NewRESTStorage method), caused the nil panic.

So it is better to check the nil and return err msg (this PR), or just keep the panic?

Reply:
I think it's reasonable for this to assume it's inputs are non-nil and panic if they are